src/02-common-principles.md

```diff
@@ -383,6 +383,200 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
+and the BIDS URIs for the raw dataset (excluding `/derivatives`) MUST be resolved with respect to `/dataset_description.json`.
+
+However if a derivatives dataset is nested under a raw dataset and only the raw dataset has a `dataset_description.json` file,
+all BIDS URIs MUST be resolved with respect to the `/dataset_description.json` file.
+
+(Assuming that in the two above examples `/` refers to the root of the BIDS raw dataset with a nested derivatives dataset.)
+
+### Refer to a file within a dataset
+
+To link for example to a derivative file that is stored in a nested `derivatives/`
+directory within a dataset:
+`bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv1": "file://derivatives/derivative1"
+    }
+
+}
+```
+
+Alternatively, we can also make use of the reserved `<dataset-name>`: `""` (an empty string),
+which always points to the root of the current dataset, to point to the same file using the following syntax:
+`bids::/derivatives/derivative1/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+Thus, for this specific case there are two ways to refer to the same file:
+
+1.  `bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`,
+    with `deriv1` being specified within `dataset_description.DatasetLinks`,
+    as shown in the example above.
+
+1.  `bids::/derivatives/derivative1/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`,
+    with `""` (an empty string) NOT being a key in `dataset_description.DatasetLinks`,
+    because it is a reserved value that always points to the root of the current dataset.
+
+**If *all* BIDS URIs in a dataset *only* use an empty string (`""`) as a `<dataset-name>`,**
+**the `DatasetLinks` metadata MAY not be specified.**
+
+### Refer to a remote dataset
+
+To link for example to a derivative file that is stored in a `derivatives/`
+directory that is stored on a remote server called `mydatahost.com`:
+`bids:deriv2:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv2": "https://mydatahost.com/derivatives/derivative2"
+    }
+}
+```
+
+This means that the needed files can be obtained by downloading the data
+from `https://mydatahost.com/derivatives/derivative2`, and then navigating
+to `/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz` relative to that
+downloaded data.
+
+However,
+arbitrary remote server locations such as the `mydatahost.com` example server may not be stable.
+Therefore it is RECOMMENDED to instead point to archived versions of a datasets using their DOIs:
+
+```json
+{
+    "DatasetLinks": {
+        "deriv3": "doi:10.18112/openneuro.ds003669.v1.0.0"
+    }
+}
+```
+
+Again, this means that the linked dataset must be downloaded from the DOI address,
+and the file will be found by navigating to the location specified in the BIDS URI
+relative to the downloaded data.
+
+### Refer to a file outside of a dataset but on the same host
```

Comment:
since this is the 4th example, can you use `deriv4` rather than `deriv3`?

Reply:
yes, makes sense --> https://github.com/bids-standard/bids-specification/pull/820/commits/475b4ca345ae15760e78651cbd86b4fab1d93b32